### PR TITLE
fixed css issues

### PR DESCRIPTION
**Pull request by eikaramba**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { autoinject } from "aurelia-framework";
 import { RouterConfiguration, Router } from "aurelia-router";
+import {PLATFORM} from 'aurelia-pal';
 import { I18N } from 'aurelia-i18n';
+import 'app.scss';
```
Comment (eikaramba) [inline src/app.ts:5]: this was needed, as the require in the html file was not working. see webpack config

File: src/main.ts [hunk 2]
```diff
@@ -38,5 +40,5 @@ export function configure(aurelia: Aurelia) {
       });
     });
   
-  aurelia.start().then(() => aurelia.setRoot());
+  aurelia.start().then(() => aurelia.setRoot(PLATFORM.moduleName('app')));
```
Comment (eikaramba) [inline src/main.ts:43]: this was crucial

File: webpack.config.js [hunk 3]
```diff
@@ -54,7 +53,7 @@ module.exports = ({production, server, extractCss, coverage} = {}) => ({
       // CSS required in JS/TS files should use the style-loader that auto-injects it into the website
       // only when the issuer is a .js/.ts file, so the loaders are not applied inside html templates
       {
-        test: /\.css$/i,
+        test: /\.(scss|css)$/i,
```
Comment (eikaramba) [inline webpack.config.js:56]: as an app.scss is used, this was needed too

File: webpack.config.js [hunk 4]
```diff
@@ -90,6 +89,71 @@ module.exports = ({production, server, extractCss, coverage} = {}) => ({
   },
   plugins: [
     new AureliaPlugin(),
+    new ModuleDependenciesPlugin({
```
Comment (eikaramba) [inline webpack.config.js:92]: this is not needed i guess, just wanted to keep it there for now

File: src/resources/index.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
+import {PLATFORM} from 'aurelia-pal';
 
 export function configure(config: FrameworkConfiguration) {
   config.globalResources([
-    './attributes/md-auto-focus',
-    './elements/nav-bar',
-    './elements/modals/add-todo-modal'
+    PLATFORM.moduleName('./attributes/md-auto-focus'),
+    PLATFORM.moduleName('./elements/nav-bar'),
+    PLATFORM.moduleName('./resources/elements/modals/add-todo-modal')
```
Comment (Thanood) [inline src/resources/index.ts:8]: Did I leave another `resources` folder in `resources` or why is this necessary?

File: webpack.config.js (hunk 4 above)
Comment (Thanood) [reply webpack.config.js:92]: Well, I'd like to try without. 😃 
Since somehow this is the point of it all.. We can comment it and leave it in, of course.

File: src/main.ts [hunk 6]
```diff
@@ -2,6 +2,8 @@ import {Aurelia} from 'aurelia-framework'
 import {PLATFORM} from 'aurelia-pal';
 import {I18N, Backend} from 'aurelia-i18n';
 import environment from './environment';
+import 'materialize-css';
+import 'aurelia-materialize-bridge';
```
Comment (Thanood) [inline src/main.ts:6]: I'm curious: Is the bridge not imported by using it as a plugin in `main.ts`?

**Comment by Thanood** [comment]
These are a lot of fixes, thanks. 👍 
I'm just wondering which of these changes fixes the bridge's css problem. Or is that a follow-up error of my many other errors. 😃

**Comment by eikaramba** [comment]
sry for the mess, i was in a hurry and just commited everything i changed. i really should have commited after every "step" of fixing more problems. indeed some stuff is not needed, but tbh i wanted to work my way back from a working state and just remove everything that might not be needed ;)

anyway for the css issue see this comment in the other issue:

> remove this line https://github.com/Thanood/md-todo-webpack/blob/master/webpack.config.js#L22, it causes problems as CSS required in templates cannot be extracted safely because Aurelia would try to require it again in runtime.

I don't know 100% why it is working after removing it, and you said that you tried that and it didn't worked. That is strange, but it actually fixed the problem for me.

File: src/main.ts (hunk 6 above)
Comment (eikaramba) [reply src/main.ts:6]: you are right, this might not be needed. that is just one of the "better have it there and if everything works, let's try without it" cases. I just have this code in my repo, which itself is working fine. So i thought.... could not harm :P

File: src/resources/index.ts (hunk 5 above)
Comment (eikaramba) [reply src/resources/index.ts:8]: my bad.... the resources folder can be removed in the path.

related to the PLATFORM change: I thought the problem of aurelia not finding add-todo-modal could be related to this. And according to https://github.com/jods4/aurelia-webpack-build/wiki/Managing-dependencies this needs to be done. However it did not help (but also did not harm and might have resulted in other problems later)

**Comment by Thanood** [comment]
Well, tbh it didn't fix anything when I removed that line, hmm.. I'm confused.
I'll try with your changes now.

**Comment by Thanood** [comment]
Okay, I had this `index.html` in my root which was loaded somehow. After deleting it, I've got the same error `Error: No Aurelia APIs are defined for the element: "ADD-TODO-MODAL".`

**Comment by Thanood** [comment]
After loading resources again and changing it to the recommended approach ([see here](https://github.com/Thanood/md-todo-webpack/blob/master/src/main.ts#L18)) and removing the `resources` path in `src/resources/index.ts`, the app loads and so the bridge works.

I didn't remove/comment out `ModuleDependenciesPlugin` yet but I will try that next.

Unrelated to bridge-business, `aurelia-i18n` doesn't seem to work and the app shows translation keys everywhere. But Materialize works. 😃 
I guess the `json` files containing translations are not bundled, maybe try a different loader. I'm using aurelia-loader atm.

**Comment by Thanood** [comment]
And it still works after removing `ModuleDependenciesPlugin` ✨

**Comment by eikaramba** [comment]
wuhu, perfect! glad that i was able to help. once you publish the new new version i will also try it in my project and should it work, i think the original issue can be closed. However i guess some documentation could be added or changed in the bridge docs, maybe also or at least a link to this repo.

**Comment by Thanood** [comment]
I'll keep the issue open as a reminder.
Also, it would be nice to get i18n to work before linking this repo. Or even get rid of i18n since the most important stuff is webpack compatibility.

Big thanks again! 😃

**Comment by Thanood** [comment]
Also, there must be a way around these `require` problems...

File: src/app.ts (hunk 1 above)
Comment (Thanood) [reply src/app.ts:5]: @eikaramba any idea if `<require from="app.scss">` would work?

I will try it myself later.

Comment (eikaramba) [reply src/app.ts:5]: I don't think it will work, i tried it before i think. but go ahead, i don't have the code currently checked out so i cannot tell for sure.
